src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -680,11 +681,13 @@ private static boolean shouldCheckIdentTokenNestedUnderDot(DetailAST dotAst) {
      * @param variablesStack stack of all the relevant variables in the scope
      */
     private static void checkIdentifierAst(DetailAST identAst, Deque<VariableDesc> variablesStack) {
-        for (VariableDesc variableDesc : variablesStack) {
-            if (identAst.getText().equals(variableDesc.getName())
-                    && !isLeftHandSideValue(identAst)) {
-                variableDesc.registerAsUsed();
-                break;
+        if (identAst != null) {
```

Comment:
it is weird that null come here,
can you move this condition to ` private static void visitDotToken(DetailAST dotAst, Deque<VariableDesc> variablesStack) {` to execute `checkIdentifierAst` only when target is not null.

Reply:
done